core/src/mender-artifact.c

```diff
@@ -1152,6 +1152,20 @@ artifact_read_data_prepare(mender_artifact_ctx_t *ctx, mender_artifact_download_
         return MENDER_FAIL;
     }
 
+    size_t file_name_length = strlen(ctx->file.name);
+    if ((strlen("data/xxxx.tar.xx") == file_name_length) || (strlen("data/xxxx.tar.xxx") == file_name_length)) {
+        /*
+        * We allow compressed files _inside_ a payload:
+        *   'data/0000.tar/compressed.tar.gz'
+        * But not a compressed payload:
+        *   'data/0000.tar[.gz|.xz|.zst]'
```

Comment:
I feel like we should implement this logic in a better way than just checking lengths of strings.

Reply:
We do also check if it's compressed, though. Asserting the length is just a way of ensuring that we don't also catch files like `data/0000.tar/compressed.tar.gz`. The files will start with e.g. `data/000.tar.` either way, so shouldn't checking if the size is `data/0000.tar[.gz|.xz|.zst]`, and then checking the suffix be sufficient? Do you perhaps have another suggestion

Reply:
I think it's fine, but there should be comment explaining why checking the length only is good enough in this particular place.

Reply:
Just want to make sure that you know that it is also checked  with `is_compressed` below the comment, we don't only check the length - we also check the suffix of the filename